charts/core/src/Chart/hooks/useChart.ts

```diff
@@ -0,0 +1,124 @@
+import { useEffect, useRef, useState } from 'react';
+import { LineChart as EchartsLineChart } from 'echarts/charts';
+import {
+  GridComponent,
+  LegendComponent,
+  TitleComponent,
+  ToolboxComponent,
+  TooltipComponent,
+} from 'echarts/components';
+import * as echarts from 'echarts/core';
+import { CanvasRenderer } from 'echarts/renderers';
+import debounce from 'lodash.debounce';
+
+import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
+import { DarkModeProps } from '@leafygreen-ui/lib';
+
+import { ChartOptions, SeriesOption } from '../../Chart/Chart.types';
+import { colors } from '../colors';
+import { getDefaultChartOptions } from '../config';
+
+import { addSeries, removeSeries, updateOptions } from './updateUtils';
+
+/**
+ * Register the required components. By using separate imports, we can avoid
+ * importing the entire echarts library which will reduce the bundle size.
+ * Must be added to if additional functionality is supported.
+ */
+echarts.use([
+  TitleComponent,
+  TooltipComponent,
+  GridComponent,
+  LegendComponent,
+  EchartsLineChart,
+  CanvasRenderer,
+  ToolboxComponent,
+]);
+
+interface ChartHookProps extends DarkModeProps {
+  onChartReady?: () => void;
+}
+
+/**
+ * Creates a generic Apache ECharts options object with default values for those not set
+ * that are in line with the designs and needs of the design system.
+ */
+export function useChart({
+  darkMode: darkModeProp,
+  onChartReady,
+}: ChartHookProps) {
+  const chartRef = useRef(null);
+  const chartInstanceRef = useRef<echarts.EChartsType | undefined>();
+  const { theme } = useDarkMode(darkModeProp);
+  const [chartOptions, setChartOptions] = useState(
+    getDefaultChartOptions(theme),
+  );
+
+  useEffect(() => {
+    const chartInstance = echarts.init(chartRef.current);
+    chartInstanceRef.current = chartInstance;
+    chartInstance.setOption(chartOptions);
+
+    // ECharts does not automatically resize when the window resizes.
+    const resizeHandler = () => {
+      chartInstance.resize();
+    };
+    window.addEventListener('resize', resizeHandler);
+
+    if (onChartReady) {
+      chartInstance.on('finished', onChartReady);
+    }
+
+    return () => {
+      window.removeEventListener('resize', resizeHandler);
+      chartInstance.dispose();
+    };
+  }, []);
+
+  const updateChartRef = debounce((chartOptions: Partial<ChartOptions>) => {
+    chartInstanceRef.current?.setOption(chartOptions);
+  }, 50);
+
+  const addChartSeries = (data: SeriesOption) => {
+    setChartOptions(currentOptions => {
+      const updatedOptions = addSeries(currentOptions, data);
```

Comment:
I still think using a reducer would be a bit better, but not significant enough to change this